source/type/message.cpp

```diff
@@ -185,6 +191,130 @@ std::vector<ImpactedVJ> get_impacted_vehicle_journeys(const LineSection& ls,
+                return true;
+            }
+
+            // Filtering each journey to see if it's impacted by the section.
+            auto section = vj.get_sections_ranks(rs.start_point, rs.end_point);
+            // If the vj pass by both stops both elements will be different than nullptr, otherwise
+            // it's not passing by both stops and should not be impacted
+
+            if ((!section.empty() && rs.blocked_stop_areas.empty())
+                || (!section.empty() && blocked_sa_sequence_matching(blocked_sa_uri_sequence, vj, section))) {
+                // Once we know the line section is part of the vj we compute the vp for the adapted_vj
+                nt::ValidityPattern new_vp{vj.validity_patterns[rt_level]->beginning_date};
+                for (const auto& period : impact.application_periods) {
+                    // get the vp of the section
+                    new_vp.days |= vj.get_vp_for_section(section, rt_level, period).days;
+                }
+                // If there is effective days for the adapted vp we're keeping it
+                if (!new_vp.days.none()) {
+                    LOG4CPLUS_TRACE(log, "vj " << vj.uri << " is affected, keeping it.");
+                    new_vp.days >>= vj.shift;
+                    vj_vp_pairs.emplace_back(vj.uri, new_vp, std::move(section));
+                }
+            }
+            return true;
+        });
+    }
+    return vj_vp_pairs;
+}
+
+// convert URI SA rail section into ordered list
+BlockedSAList create_blocked_sa_sequence(const RailSection& rs) {
+    BlockedSAList blocked_sa_uri_sequence;
+
+    if (rs.start_point == nullptr || rs.end_point == nullptr) {
+        return blocked_sa_uri_sequence;
+    }
+
+    // add start_point SA
+    blocked_sa_uri_sequence.insert(std::make_pair(0, rs.start_point->uri));
+    // add blocked_stop_areas list
+    for (const auto bsa : rs.blocked_stop_areas) {
+        blocked_sa_uri_sequence.insert(std::make_pair(bsa.second + 1, bsa.first));
+    }
+    // add end_point SA
+    if (blocked_sa_uri_sequence.rbegin() != blocked_sa_uri_sequence.rend()) {
+        blocked_sa_uri_sequence.insert(std::make_pair(blocked_sa_uri_sequence.rbegin()->first + 1, rs.end_point->uri));
+    }
+    return blocked_sa_uri_sequence;
+}
+
+// Only check if all blocked SA are contained inside route
+bool is_route_to_impact_content_sa_list(const BlockedSAList& blocked_sa_uri_sequence,
+                                        const boost::container::flat_set<StopArea*>& stop_area_list) {
+    for (const auto sa : blocked_sa_uri_sequence) {
+        std::string bsa_uri = sa.second;
+        auto result = std::find_if(stop_area_list.begin(), stop_area_list.end(),
+                                   [bsa_uri](const auto& sa) { return sa->uri == bsa_uri; });
+        if (result == stop_area_list.end()) {
+            return false;
+        }
+    }
+    return true;
```

Comment:
Use `const auto& sa` to avoid copy pair<int,string>
and in lambda use `[&bsa_uri] ` for the same reason